### ISSUE TITLE
[Usability Enhancement] un-export `local_operator` from `netket.operator`

### ISSUE THREAD

**Issue by PhilipVinc** [netket/netket#445]
Confuses users between `local_operator`and`LocalOperator`.
It's not supposed to be part of the interface

**Comment by gcarleo** [comment]
Yes maybe we should just rename local_operator into local_estimator or something... better option is to just hide it in the generic implementation of our chimera class implementing a ```VariationalState```!

**Comment by PhilipVinc** [comment]
In the future yes, I 100% agree, but to quickly address this issue I think we only need to rename the file `local_operator` to `_local_operator`.

**Comment by gcarleo** [comment]
agreed...

**Comment by femtobit** [comment]
> Yes maybe we should just rename local_operator into local_estimator or something... better option is to just hide it in the generic implementation of our chimera class implementing a VariationalState!

I think this comment refers to `local_values` (i.e., the function computing the local energy/values of operators), not `local_operator`, right?

In this case, I'd just like to add that I'd also be in support of finding a better name for `local_values` (maybe as part of the `VariationalState` class). At least in my experience, the term "local values" is generic enough (especially compared to "local energy") that it can be pretty confusing when explaining VMC. `local_estimator` would already be an improvement IMHO.

**Comment by PhilipVinc** [comment]
No, my comment is about `local_operator`.
If I try to tab complete `nk.operator` I get
```python
>>> netket.operator.
netket.operator.AbstractOperator(     netket.operator.PauliStrings(         netket.operator.hamiltonian
netket.operator.BoseHubbard(          netket.operator.abstract_operator     netket.operator.local_liouvillian
netket.operator.GraphOperator(        netket.operator.bose_hubbard          netket.operator.local_operator
netket.operator.Heisenberg(           netket.operator.boson                 netket.operator.local_values(
netket.operator.Ising(                netket.operator.der_local_values(     netket.operator.pauli_strings
netket.operator.LocalLiouvillian(     netket.operator.der_local_values_jax  netket.operator.spin
netket.operator.LocalOperator(        netket.operator.graph_operator
```

I think we should be hiding `local_operator` (the file).

**Comment by femtobit** [comment]
Yes, I get that and I agree. I was referring to Giuseppe's comment.

**Comment by femtobit** [comment]
Should be resolved by #454.
